frame/oracle/src/lib.rs

```diff
@@ -281,325 +286,93 @@ pub mod pallet {
-                  )
-                  .is_err()
-                  {
-                    // FIXME: Add rollback
-                  }
-
-                  // 12. c) Register a new trading fees associated with the account.
-                  // A percentage of the network profits will be re-distributed to the account at the end of the era.
-                  T::Fees::register_swap_fees(
-                    market_maker_trade_intent.account_id.clone(),
-                    trade.token_to,
-                    mm.amount_to_send,
-                    market_maker_trade_intent.swap_type.clone(),
-                    market_maker_trade_intent.is_market_maker,
-                  )
-                  .map_err(|_| Error::<T>::MarketMakerSwapFeeRegistrationFailed)?;
-
-                  // 13. Emit market maker trade event on chain
-                  Self::deposit_event(Event::<T>::SwapProcessed {
-                    request_id: mm.request_id,
-                    initial_extrinsic_hash: market_maker_trade_intent.extrinsic_hash,
-                    status: market_maker_trade_intent.status.clone(),
-                    account_id: market_maker_trade_intent.account_id.clone(),
-                    currency_from: market_maker_trade_intent.token_from,
-                    currency_amount_from: mm.amount_to_send,
-                    currency_to: market_maker_trade_intent.token_to,
-                    currency_amount_to: mm.amount_to_receive,
-                  });
-
-                  // 14. Delete the intent if it's completed or if it's a market order
-
-                  // release order if its within slippage values
-                  if market_maker_trade_intent.status == SwapStatus::Completed
-                    || market_maker_trade_intent.swap_type == SwapType::Market
-                  {
-                    Self::try_delete_account_swap(
-                      &market_maker_trade_intent.account_id,
-                      mm.request_id,
-                    )
-                    .map_err(|_| Error::<T>::DeleteMarketMakerSwapFailed)?;
-                    Self::swap_release_funds(market_maker_trade_intent)
-                      .map_err(|_| Error::<T>::ReleaseMarketMakerUnswappedFundsFailed)?;
-                    *mm_trade_request = None;
-                  } else {
-                    Self::try_update_account_swap_status(
-                      &market_maker_trade_intent.account_id,
-                      mm.request_id,
-                      market_maker_trade_intent.status.clone(),
-                    )
-                    .map_err(|_| Error::<T>::UpdateMarketMakerAccountSwapRequestStatusFailed)?;
-                  }
-                }
-
-                Ok(())
-              })?;
-            }
+      let mut trade_total_from: Balance = 0;
+      let mut trade_total_to: Balance = 0;
+      let mut trade_latest_from_filled: Balance = trade.amount_from_filled;
+      let mut trade_latest_to_filled: Balance = trade.amount_to_filled;
```

Comment:
Took me a while to parse to utility of these; they could be named a lot better but even so, they seem entirely superfluous as you have `trade_total_from` and `trade_total_to`?

Reply:
Done